sources/packages/api/src/route-controllers/student-account-applications/student-account-application.aest.controller.ts

```diff
@@ -0,0 +1,174 @@
+ */
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("student-account-application")
+@ApiTags(`${ClientTypeBaseRoute.AEST}-student-account-application`)
+@Injectable()
+export class StudentAccountApplicationAESTController extends BaseController {
+  constructor(
+    private readonly formService: FormService,
+    private readonly studentAccountApplicationsService: StudentAccountApplicationsService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get the summary list of all student account applications
+   * waiting to be assessed by the Ministry.
+   * @returns summary list of pending student account applications.
+   */
+  @Get("pending")
+  async getPendingStudentAccountApplications(): Promise<
+    StudentAccountApplicationSummaryAPIOutDTO[]
+  > {
+    const accountApplications =
+      await this.studentAccountApplicationsService.getPendingStudentAccountApplications();
+    return accountApplications.map((accountApplication) => ({
+      id: accountApplication.id,
+      fullName: getUserFullName(accountApplication.user),
+      submittedDate: accountApplication.submittedDate,
+    }));
+  }
+
+  /**
+   * Get the student account application previously submitted
+   * by the student for the basic BCeID student account creation.
+   * @param studentAccountApplicationId student account application id.
+   * @returns student account application.
+   */
+  @ApiNotFoundResponse({
+    description: "Student account application not found.",
+  })
+  @Get(":studentAccountApplicationId")
+  async getStudentAccountApplicationById(
+    @Param("studentAccountApplicationId", ParseIntPipe)
+    studentAccountApplicationId: number,
+  ): Promise<StudentAccountApplicationAPIOutDTO> {
+    const accountApplication =
+      await this.studentAccountApplicationsService.getStudentAccountApplicationsById(
+        studentAccountApplicationId,
+      );
+    if (!accountApplication) {
+      throw new NotFoundException("Student account application not found.");
+    }
+    return {
+      id: accountApplication.id,
+      submittedData: accountApplication.submittedData,
+    };
+  }
+
+  /**
+   * Approve the student account application associating the user
+   * with a student account. The Ministry can also adjust any student
+   * data that will then be used to create the student account.
+   * @param studentAccountApplicationId student account application id.
+   * @returns new student id created as a result of the approval.
+   */
+  @ApiNotFoundResponse({
+    description: "Student account application not found.",
+  })
+  @Roles(Role.StudentApproveDeclineAccountRequests)
+  @Post(":studentAccountApplicationId/approve")
+  async approveStudentAccountApplication(
+    @Param("studentAccountApplicationId", ParseIntPipe)
+    studentAccountApplicationId: number,
+    @Body() payload: StudentAccountApplicationApprovalAPIInDTO,
+    @UserToken() userToken: IUserToken,
+  ): Promise<void> {
```

Comment:
Post not returning the primary identifier. But comment mentions that student id is returned.

Reply:
Chnaged.